### PR TITLE
fix: add select sidebar flatten

### PR DESCRIPTION
**Pull request by davidmenendez**
Contributes to #1618 

reworks the sidebar to use the carbon `Accordion` to display selected items and adds a utility function to flatten the list of items to make nested items easily searchable.

### CODE REVIEW

File: packages/cloud-cognitive/src/components/AddSelect/AddSelectSidebar.js [hunk 1]
```diff
@@ -27,23 +40,32 @@ export let AddSelectSidebar = ({
...
-      </div>
+        </div>
+      )}
     </div>
   );
 };
 
 AddSelectSidebar.propTypes = {
   influencerTitle: PropTypes.string,
+  items: PropTypes.array,
```
Comment (matthewgallo) [inline packages/cloud-cognitive/src/components/AddSelect/AddSelectSidebar.js:68]: It also might be worth extending the prop types for `items` a bit now that it looks like an `id` property is required

ie:
```jsx
items: PropTypes.arrayOf(
  PropTypes.shape({
    value: PropTypes.string,
    id: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
  })
)
```

I don't know if those are the correct prop types but giving a bit more depth as to the structure of `items` might be good.

Comment (davidmenendez) [reply packages/cloud-cognitive/src/components/AddSelect/AddSelectSidebar.js:68]: that's fair 👍

**Comment by davidmenendez** [comment]
@matthewgallo updated the base `AddSelect` component to require required props in the items array

File: packages/cloud-cognitive/src/components/AddSelect/AddSelectSidebar.js [hunk 2]
```diff
@@ -27,23 +40,32 @@ export let AddSelectSidebar = ({
...
-      <div className={`${blockClass}-body`}>
-        {multiSelection.length > 0 ? (
-          multiSelection.map((item) => <p key={item}>{item}</p>)
-        ) : (
+      {multiSelection.length > 0 ? (
+        <Accordion align="start">
+          {sidebarItems.map((item) => (
+            <AccordionItem title={item.value} key={item.id}>
+              {Object.keys(item).map((key) => (
+                <p key={key}>{`${key} - ${item[key]}`}</p>
```
[the remark this answers is not in the document]
Comment (davidmenendez) [reply packages/cloud-cognitive/src/components/AddSelect/AddSelectSidebar.js:48]: this is just temporary. i can make another update to this to clean it up.

**Review by matthewgallo — DISMISSED** [review]

**Comment by davidmenendez** [comment]
@matthewgallo also updated the sidebar string issue

**Pull request merged**